tests/eas/st_hold.py

```diff
@@ -0,0 +1,408 @@
+    NO_SCHEDTUNE_BOOST_HOLD_ALL is the sched_feature enabled, and
+    this means that only rt-tasks in boosted groups trigger the
+    boost hold mechanism. If SCHEDTUNE_BOOST_HOLD_ALL is enabled
+    instead, any rt or cfs task in a boosted group will trigger
+    the boost holding.
+    This test uses a 40ms periodic 1% cfs task in a 100% boosted
+    group and verifies that schedutil does not raise the cpu
+    frequency to max for the entire duration of the test. This
+    test is skipped if SCHEDTUNE_BOOST_HOLD_ALL is set.
+    """
+    task_name = "1pct_100"
+    cgroup_name = BOOSTED_CGROUP_NAME
+
+    @classmethod
+    def setUpClass(self, *args, **kwargs):
+        super(NormalBoostedTaskTest, self).setUpClass(*args, **kwargs)
+
+    @classmethod
+    def populate_params(self):
+        self.rtapp = RTA(self.target,self.rta_name)
+        self.rtapp.conf(
+            kind='profile',
+            params={
+                self.task_name: Periodic(
+                    period_ms=40, duty_cycle_pct=1,
+                    duration_s=2,
+                    cpus=self.test_cpu,
+                ).get(),
+            },
+            run_dir='/tmp'
+        )
+
+    @classmethod
+    def child_check_config(self):
+        self.check_hold_config()
+
+    def test_cpu_frequency(self):
+        self._test_cpu_frequency()
+
+class NormalBoostedTaskTestBoostAll(_SchedtuneHoldTest):
+    """
+    Schedtune Boost Hold keeps the boost level active for at least
+    50ms since the last enqueue, for qualifying tasks. Eligibility is
+    controlled by a sched_feature. In the default configurations,
+    NO_SCHEDTUNE_BOOST_HOLD_ALL is the sched_feature enabled, and
+    this means that only rt-tasks in boosted groups trigger the
+    boost hold mechanism. If SCHEDTUNE_BOOST_HOLD_ALL is enabled
+    instead, any rt or cfs task in a boosted group will trigger
+    the boost holding.
+    This test uses a 40ms periodic 1% cfs task in a 100% boosted
+    group and verifies that schedutil raises the cpu frequency
+    to max for the entire duration of the test. This
+    test is skipped if NO_SCHEDTUNE_BOOST_HOLD_ALL is set.
+    """
+    task_name = "1pct_100a"
+    cgroup_name = BOOSTED_CGROUP_NAME
+
+    @classmethod
+    def setUpClass(self, *args, **kwargs):
+        super(NormalBoostedTaskTestBoostAll, self).setUpClass(*args, **kwargs)
+
+    @classmethod
+    def populate_params(self):
+        self.rtapp = RTA(self.target,self.rta_name)
+        self.rtapp.conf(
+            kind='profile',
+            params={
+                self.task_name: Periodic(
+                    period_ms=40, duty_cycle_pct=1,
+                    duration_s=2,
+                    cpus=self.test_cpu,
+                ).get(),
+            },
+            run_dir='/tmp'
+        )
+
+    @classmethod
+    def child_check_config(self):
+        self.check_hold_config(want_default=False)
+
+    def test_cpu_frequency(self):
```

Comment:
Should probably be a classmethod like all the other ones

Reply:
Given that all classes have this method, it should be implemented in the base class, and overriden only when needed to add the extra parameter

Reply:
ok